### PR TITLE
Add contents and packages checks for .yml  [Quick review]

### PR DESCRIPTION
**Pull request by Kurlov**
This PR add a BuildException raise in case neither `contents` nor `packages` node provided

### CODE REVIEW

File: compiler/quilt/tools/build.py [hunk 1]
```diff
@@ -527,6 +527,8 @@ def load_yaml(filename, optional=False):
         data = fd.read()
     try:
         res = yaml.load(data)
+        if not any(k in res.keys() for k in ['contents', 'packages']):
+            raise BuildException('"contents" or "packages" node must be specified')
```
Comment (akarve) [inline compiler/quilt/tools/build.py]: Let's break this out into two cases:
* `Error in FILENAME: missing 'contents' node`
* `Error in FILENAME: missing 'pacakges' node`

Comment (akarve) [reply compiler/quilt/tools/build.py]: Also note, 'packages' is an entirely different use case. That's only for `quilt.yml`.

File: compiler/quilt/test/build_no_contents_node.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+does_not:
+  contain: contents.node
```
Comment (akarve) [inline compiler/quilt/test/build_no_contents_node.yml]: for proper test coverage we should have a build file without 'contents', a `quilt.yml` file without 'package'

Comment (Kurlov) [reply compiler/quilt/test/build_no_contents_node.yml]: `packages` exception tests in `test_command.py:test_install_dependencies_from_invalid_file` where temporary `.yml` created.

**Comment by Kurlov** [comment]
I split checks for `packages` and `contents`. I think lack of `packages` is more about `command.install` error, thus it's `CommandException`. 
Please, tell me if I get it wrong way.

File: compiler/quilt/test/test_build.py [hunk 3]
```diff
@@ -264,6 +264,16 @@ def test_build_yaml_syntax_error(self):
         with assertRaisesRegex(self, build.BuildException, r'Bad yaml syntax.*build_bad_syntax\.yml'):
             build.build_package(None, 'test_syntax_error', PACKAGE, path)
 
+    def test_build_no_contents_node(self):
+        """
+        Attempt to build a yml file without contents node nor packages
```
Comment (eode) [inline compiler/quilt/test/test_build.py]: Docstring inaccurate:  You tested 'packages' elsewhere

**Pull request merged**